zephyr/include/rtos/cache.h

```diff
@@ -8,12 +8,59 @@
 
 /* TODO: align with Zephyr generic cache API when ready */
 #define __SOF_LIB_CACHE_H__
+
+#if !CONFIG_ARM64
 #include <arch/lib/cache.h>
+#else
+#include <zephyr/cache.h>
+#endif /* CONFIG_ARM64 */
 
 /* writeback and invalidate data */
 #define CACHE_WRITEBACK_INV	0
 
 /* invalidate data */
 #define CACHE_INVALIDATE	1
 
+#if CONFIG_ARM64
+/* For now, run-time data cache line size detection
+ * is not supported by SOF as the data cache line size
+ * needs to be known at compilation time because of
+ * dependencies such as the alignment of some data
+ * (e.g: the heap).
+ */
+#if !defined(CONFIG_DCACHE_LINE_SIZE_DETECT) && \
+	(CONFIG_DCACHE_LINE_SIZE > 0)
+
+#define DCACHE_LINE_SIZE CONFIG_DCACHE_LINE_SIZE
+
+static inline void dcache_writeback_region(void *addr, size_t size)
+{
+	/* TODO: return value should probably be checked here */
+	sys_cache_data_flush_range(addr, size);
+}
```

Comment:
Hmm, did you consider to put this to arch/arm64/lib/cache.h ? I think that would be cleaner still as long as we need some wrapping on SOF side? 

Given our intention is to remove rtos/cache.h and use sys_cache_.* directly, this is probably not a big issue to have some ifdefs in this file, but if it's not a big task, why not use the existing infra and add arm64 arch files to SOF. I think you'll need that for other stuff as well (like the compiler_version.h thing).

Reply:
Moving all this stuff to `arch/arm64/lib/cache.h` or keeping it in here should be the same thing as it's only supposed to be temporary anyways (thanks to @nashif's patches which hopefully will take care of things on xtensa side). Also, shouldn't we be trying to get rid of the `arch/` hierarchy entirely instead of adding more stuff to it? (although only temporarily in this case)

Reply:
@LaurentiuM1234  Right, I change my mind on the compiler_info and given this will go away as well, no need to add arch if we already know now it will be removed soon.